pkg/storage/unified/resource/metrics.go

```diff
@@ -0,0 +1,41 @@
+package resource
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	once                 sync.Once
+	StorageServerMetrics *StorageApiMetrics
+)
+
+type StorageApiMetrics struct {
+	OptimisticLockFailed *prometheus.CounterVec
+}
+
+func NewStorageMetrics() *StorageApiMetrics {
+	once.Do(func() {
+		StorageServerMetrics = &StorageApiMetrics{
+			OptimisticLockFailed: prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Namespace: "resource_storage",
+					Name:      "optimistic_lock_failed",
+					Help:      "count of optimistic locks failed",
+				},
+				[]string{"action"},
+			),
+		}
+	})
+
+	return StorageServerMetrics
+}
+
+func (s *StorageApiMetrics) Collect(ch chan<- prometheus.Metric) {
+	s.OptimisticLockFailed.Collect(ch)
+}
+
+func (s *StorageApiMetrics) Describe(ch chan<- *prometheus.Desc) {
+	s.OptimisticLockFailed.Describe(ch)
+}
```

Comment:
Do these make sense? Sooner or later we're going to add another metric, then we'll either need to make more wrappers or switch callers from `s.Collect(ch)` to `s.OptimisticLockFailed.Collect(ch)`

Reply:
This was copied directly from https://github.com/grafana/grafana/blob/main/pkg/services/store/entity/sqlstash/metrics.go#L18

I don't have any opinion on what the best approach is -- we can remove it or replace with something better